### PR TITLE
cpio: on extract, honor unix bits like SUID, GUID, and sticky

### PR DESCRIPTION
**Pull request by rminnich**
Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

### CODE REVIEW

File: pkg/cpio/fs_unix.go [hunk 1]
```diff
@@ -58,6 +58,21 @@ func setModes(r Record) error {
...
 
+func toFileMode(r Record) os.FileMode {
+	m := os.FileMode(perm(r))
+	if r.Mode&syscall.S_ISUID != 0 {
+		m |= os.ModeSetuid
+	}
+	if r.Mode&syscall.S_ISGID != 0 {
+		m |= os.ModeSetgid
+	}
+	// Where is sticky defined?
```
Comment (hugelgupf) [inline pkg/cpio/fs_unix.go]: S_ISVTX. Unix package probably, maybe forgotten in syscall?

**Comment by rminnich** [comment]
tested now on nichrome.

File: pkg/cpio/fs_unix.go [hunk 2]
```diff
@@ -6,6 +6,7 @@ package cpio
 
 import (
 	"fmt"
+	"golang.org/x/sys/unix"
```
Comment (rjoleary) [inline pkg/cpio/fs_unix.go]: nit: The style is to put the non-standard packages in their own sections. For an example, see: https://github.com/u-root/u-root/blob/master/pkg/kexec/kexec_linux_amd64.go#L12

Comment (rminnich) [reply pkg/cpio/fs_unix.go]: darn it. I did not even pay attention, I just assumed gofmt did this and did not look. Fixing now.

Comment (rminnich) [reply pkg/cpio/fs_unix.go]: Although in many ways calling this a non-standard package is a bet of a stretch (to me).

Comment (rminnich) [reply pkg/cpio/fs_unix.go]: Fixed, can you take a quick look, since the one line fix reset approved :-)

**Comment by rminnich** [comment]
Also, looking at the source to MkdirAll, it does the stat for you:
	dir, err := Stat(path)
  	if err == nil {
  		if dir.IsDir() {
  			return nil
  		}
  		return &PathError{"mkdir", path, syscall.ENOTDIR}
  	}

no need for us to do the stat: it does it for us. No need to do the stat: why do it twice?

**Pull request merged**